### PR TITLE
Add CITATION.cff

### LINKED ISSUE #12779: Add a CITATION.cff to the root of the repository
### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### What is the idea?

It would be helpful for conda to have a CITATION.cff file that contains the metadata necessary for citing conda.

### Why is this needed?

While it is widely accepted that related academic research should be cited in research articles, it is less common to formally cite software. There has been a movement to more formal citation of software.

Currently, it is difficult to know how one should cite conda, since that information is not explicitly provided.

### What should happen?

Here are a few things that become possible with this addition:

- The user can read the file to create a citation in whatever format is needed for their work.
- GitHub parses the file and adds an option on the side bar (when visiting the repo on a desktop browser) to export the citation as APA or BibTeX format. BibTeX files can be imported into reference managers, so that citations can be generated in a large number of formats.
- Reference manager browser extensions such as Zotero will detect this information and can automatically add the citation to the users reference manager library.

### Additional Context

[GitHub recommends this](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files#about-citation-files) as the way to provide citation information, and provides integration for using this type of file.

This is also currently implemented by many popular packages such as [pandas](https://github.com/pandas-dev/pandas) and [tensorflow](https://github.com/tensorflow/tensorflow).

### PR DESCRIPTION
**Pull request by schackartk**
### Description

This pull request adds a CITATION.cff file to the root of the repository and would resolve #12779. As discussed in that issue, it provides several ways to help users cite conda. Hopefully this will help give conda it's due credit when it is used!

### Considerations

What information should be included in the citation file is up to the maintainers. I have done my best to fill out the information, but it can be difficult to know what to include (hence the need for such a file!)

A few things that I would suggest for further consideration:

- Who to list as authors? I see that conda does a good job of keeping updated lists. If it is possible to automate the task, it may be useful to add a full author list to the file and keep it up to date. Many projects use a broader approach like "The conda development team" or "Contributors to conda" to simplify this.
- What to use as the date? This is not required. It can be useful, but requires keeping it up to date.
- Should the version number be included in the file? Again, this is useful, but required more upkeep if it is not automated.

### Checklist - did you ...

- [X] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [X] Add / update necessary tests?
- [X] Add / update outdated documentation?

*Note*, I added it to the "Enhancements" section of the news template. Should it be that section, "Documentation", or "Other" instead?

### CODE REVIEW

**Comment by kenodegard** [comment]
@conda-bot check

**Comment by kenodegard** [comment]
pre-commit.ci autofix

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cff-version: 1.2.0
+title: "conda: A system-level, binary package and environment manager running on all major
+  operating systems and platforms."
+message: >-
+  If you use this software, please cite it using the metadata from this file.
+type: software
+authors:
+  - name: "The conda development team"
```
Comment (jezdez) [inline CITATION.cff:8]: Let's "contributors" here to acknowledge all kinds of contributions, not just code.
```suggestion
  - name: "conda contributors"
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+repository-code: "https://github.com/conda/conda"
+url: "https://docs.conda.io/projects/conda/"
+abstract: >-
+  Conda is a cross-platform, language-agnostic binary package manager. It is the package manager
+  used by Anaconda installations, but it may be used for other systems as well. Conda makes
+  environments first-class citizens, making it easy to create independent environments even for
+  C libraries. Conda is written entirely in Python, and is BSD licensed open source.
+license: BSD 3 clause
```
Comment (jezdez) [inline CITATION.cff:16]: The correct spelling of the license includes hyphens

```suggestion
license: BSD-3-clause
```

**Review by jezdez — CHANGES_REQUESTED** [review]

**Pull request merged**